meson.build

```diff
@@ -0,0 +1,53 @@
+project(
+  'pyperfdump',
+  'cpp',
+  version: '1.1',
+  meson_version: '>=1.6.1',
+  default_options: ['buildtype=release',
+                    'python.install_env=auto'],
+)
+
+project_description = 'Python Performance Dump module for PAPI'
+
+pyperfdump_headers = ['papi_utils.h', 'pyperfdump.h']
+pyperfdump_sources = ['src/dump_functions.cpp',
+                      'src/papi_utils.cpp',
+                      'src/perf_dump.cpp']
+inc = include_directories('include')
+
+py_dep = dependency('python3', required: true)
+papi_dep = dependency('papi', required: true)
+deps = [py_dep, papi_dep]
+build_args = []
+
+py = import('python').find_installation('python3')
+
+if get_option('use_mpi')
+  mpi_dep = dependency('mpi', language: 'cpp', required: true)
+  run_command(py, ['-c', 'import mpi4py'], check: true)
```

Comment:
Note in `find_installation()` you can pass `modules: ['mpi4py']` and meson will perform the import check for you. That implies re-ordering the checks so that you verify the mpi option before searching for a python installation.